autoPyTorch/pipeline/components/setup/augmentation/image/ImageAugmenter.py

```diff
@@ -0,0 +1,135 @@
+    return components
+
+
+class ImageAugmenter(BaseImageAugmenter):
+
+    def __init__(self, random_state: Optional[Union[int, np.random.RandomState]] = None):
+        super().__init__()
+        self.available_augmenters = get_components()  # type: Dict[str, BaseImageAugmenter]
+        self.random_state = random_state
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseImageAugmenter:
+        self.augmenter = iaa.Sequential([augmenter.fit(X).get_image_augmenter() for _, augmenter in
+                                         self.available_augmenters.items()])
+        return self
+
+    def transform(self, X: Dict[str, Any]) -> Dict[str, Any]:
+        X.update({'image_augmenter': self})
+        return X
+
+    def set_hyperparameters(self,
+                            configuration: Configuration,
+                            init_params: Optional[Dict[str, Any]] = None
+                            ) -> 'ImageAugmenter':
+        """
+        Applies a configuration to the given component.
+        This method translate a hierarchical configuration key,
+        to an actual parameter of the autoPyTorch component.
+
+        Args:
+            configuration (Configuration): which configuration to apply to
+                the chosen component
+            init_params (Optional[Dict[str, any]]): Optional arguments to
+                initialize the chosen component
+
+        Returns:
+            self: returns an instance of self
+        """
+        available_augmenters = get_components()
+        for name, augmenter in available_augmenters.items():
+            new_params = {}
+
+            params = configuration.get_dictionary()
+
+            for param, value in params.items():
+                if name in param:
+                    param = param.replace(name, '').replace(':', '')
+                    new_params[param] = value
+
+            if init_params is not None:
+                for param, value in init_params.items():
+                    if name in param:
+                        param = param.replace(name, '').replace(':', '')
+                        new_params[param] = value
+
+            new_params['random_state'] = self.random_state
+
+            self.available_augmenters[name] = augmenter(**new_params)
+
+        return self
+
+    @staticmethod
+    def get_hyperparameter_search_space(dataset_properties: Optional[Dict[str, str]] = None) -> ConfigurationSpace:
+        cs = ConfigurationSpace()
+
+        available_augmenters = get_components()  # type: Dict[str, BaseImageAugmenter]
+
+        if dataset_properties is None:
+            dataset_properties = dict()
+
+        # add child hyperparameters
+        for name in available_augmenters.keys():
+            preprocessor_configuration_space = available_augmenters[name].\
+                get_hyperparameter_search_space(dataset_properties)
+            cs.add_configuration_space(name, preprocessor_configuration_space)
```

Comment:
I fail to understand something -- let us say  we don't want cropping for this pipeline because it hurts performance. Is there the possibility to not have cropping? Like normally I will expect a 'no cropping' or something similar. Maybe I missed something :)

Reply:
So, the idea is for augmenters to have a probability of being applied as a hyperparameter. According to Lucas' comments, currently there is one just for RandomCutout. But it can be added for others as well.